### PR TITLE
feat: add obs documents

### PR DESCRIPTION
**Pull request by deer-hang**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: docs/obs/obs.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Huawei Cloud OBS CSI Driver
+
+The OBS CSI Driver is a CSI Specification compliant driver used by Container Orchestrators to manage
+the lifecycle of Huawei Cloud OBS Buckets.
+
+## Compatibility
+
+For sidecar version compatibility, please refer compatibility matrix for each sidecar here
+-https://kubernetes-csi.github.io/docs/sidecar-containers.html.
```
Comment (chengxiangdong) [inline docs/obs/obs.md:9]:
```suggestion
https://kubernetes-csi.github.io/docs/sidecar-containers.html.
```

File: docs/obs/obs.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+### Prerequisites
+
+- docker, kubeadm, kubelet and kubectl has been installed.
+
+### Steps
+
+- Create the config file
+
+The driver initialization depends on a [cloud config file](../../deploy/obs-csi-plugin/cloud-config).
+Make sure it's in `/etc/obs/cloud-config` on your master.
```
Comment (chengxiangdong) [inline docs/obs/obs.md:34]:
```suggestion
Create the `cloud-config` file according to [cloud config file](../../deploy/obs-csi-plugin/cloud-config) in the master node.
```

File: docs/obs/obs.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+Click to view the detailed description of the file: [cloud-config](../cloud-config.md).
```
Comment (chengxiangdong) [inline docs/obs/obs.md:36]:
```suggestion
See [cloud-config](../cloud-config.md) for configurations description.
```

File: docs/obs/obs.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+- Create secret resource
+
+```
+kubectl create secret -n kube-system generic cloud-config --from-file=/etc/obs/cloud-config
+```
+
+This should create a secret name `cloud-config` in `kube-system` namespace.
```
Comment (chengxiangdong) [inline docs/obs/obs.md:44]:
```suggestion
This will create `cloud-config` secret in `kube-system` namespace.
```

**Review by chengxiangdong — COMMENTED** [review]
there some flaws, please modify as appropriate

File: docs/obs/obs.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+- Install HuaweiCloud OBS CSI Driver
+
+```
+kubectl apply -f https://raw.githubusercontent.com/huaweicloud/huaweicloud-csi-driver/master/deploy/obs-csi-plugin/kubernetes/csi-obs-driver.yaml
+kubectl apply -f https://raw.githubusercontent.com/huaweicloud/huaweicloud-csi-driver/master/deploy/obs-csi-plugin/kubernetes/csi-obs-controller.yaml
+kubectl apply -f https://raw.githubusercontent.com/huaweicloud/huaweicloud-csi-driver/master/deploy/obs-csi-plugin/kubernetes/csi-obs-node.yaml
+```
+
+- Waiting for all the pods in running
```
[the remark this answers is not in the document]
Comment (deer-hang) [reply docs/obs/obs.md:64]: done

File: docs/obs/obs.md (hunk 1 above)
Comment (deer-hang) [reply docs/obs/obs.md:9]: done

File: docs/obs/obs.md (hunk 2 above)
Comment (deer-hang) [reply docs/obs/obs.md:34]: done

File: docs/obs/obs.md (hunk 3 above)
Comment (deer-hang) [reply docs/obs/obs.md:36]: done

File: docs/obs/obs.md (hunk 4 above)
Comment (deer-hang) [reply docs/obs/obs.md:44]: done

File: examples/obs-csi-plugin/kubernetes/dynamic/pvc.yaml [hunk 6]
```diff
@@ -1,3 +1,4 @@
+# when removing SC or PVC, the bucket resource will be cleared together
```
Comment (chengxiangdong) [inline examples/obs-csi-plugin/kubernetes/dynamic/pvc.yaml:1]:
```suggestion
# when removing SC or PVC, the bucket resources will be cleared and deleted together
```

File: docs/obs/obs-dynamic.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+# Dynamic Provisioning
+
+Dynamic Provisioning will create a new bucket. When removing SC or PVC, the bucket resource will be cleared together.
```
Comment (chengxiangdong) [inline docs/obs/obs-dynamic.md:3]:
```suggestion
Dynamic Provisioning will create a new bucket. When removing SC or PVC, the bucket resources will be cleared and deleted together.
```

File: examples/obs-csi-plugin/kubernetes/dynamic/pvc.yaml (hunk 6 above)
Comment (deer-hang) [reply examples/obs-csi-plugin/kubernetes/dynamic/pvc.yaml:1]: done

File: docs/obs/obs-dynamic.md (hunk 7 above)
Comment (deer-hang) [reply docs/obs/obs-dynamic.md:3]: done

**Pull request merged**